tests/python_tests/samples/conftest.py

```diff
@@ -0,0 +1,110 @@
+import subprocess # nosec B404
+import os
+import tempfile
+import pytest
+import shutil
+import logging
+
+# Configure logging
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+# Define model names and directories
+MODELS = {
+    "TinyLlama-1.1B-Chat-v1.0": { 
+        "name": "TinyLlama/TinyLlama-1.1B-Chat-v1.0",
+        "convert_args": []
+    },
+    "TinyLlama-1.1B-intermediate-step-1431k-3T": {
+        "name": "TinyLlama/TinyLlama-1.1B-intermediate-step-1431k-3T",
+        "convert_args": ['--trust-remote-code']
+    },  
+    "WhisperTiny": {
+        "name": "openai/whisper-tiny",
+        "convert_args": ['--trust-remote-code']
+    },
+    "open_llama_3b_v2": {
+        "name": "openlm-research/open_llama_3b_v2",
+        "convert_args": ['--trust-remote-code', '--weight-format', 'fp16']
+    }
+}
+
+TEST_FILES = {
+    "how_are_you_doing_today.wav": "https://storage.openvinotoolkit.org/models_contrib/speech/2021.2/librispeech_s5/how_are_you_doing_today.wav",
+    "adapter_model.safetensors": "https://huggingface.co/smangrul/tinyllama_lora_sql/resolve/main/adapter_model.safetensors"
+}
+
+TEMP_DIR = os.environ.get("TEMP_DIR", tempfile.mkdtemp())
+MODELS_DIR = os.path.join(TEMP_DIR, "test_models")
+TEST_DATA = os.path.join(TEMP_DIR, "test_data")
+
+SAMPLES_PY_DIR = os.environ.get("SAMPLES_PY_DIR", os.getcwd())
+SAMPLES_CPP_DIR = os.environ.get("SAMPLES_CPP_DIR", os.getcwd())
+
+# A shared fixture to hold data
+@pytest.fixture(scope="session")
+def shared_data():
+    return {}
+
+@pytest.fixture(scope="session", autouse=True)
+def setup_and_teardown(request):
+    """Fixture to set up and tear down the temporary directories."""
+    logger.info(f"Creating directories: {MODELS_DIR} and {TEST_DATA}")
+    os.makedirs(MODELS_DIR, exist_ok=True)
+    os.makedirs(TEST_DATA, exist_ok=True)
+    yield
+    if not os.environ.get("TEMP_DIR"):
+        logger.info(f"Removing temporary directory: {TEMP_DIR}")
+        shutil.rmtree(TEMP_DIR)
+    else:
+        logger.info(f"Skipping cleanup of temporary directory: {TEMP_DIR}")
+
+@pytest.fixture(scope="session")
+def convert_model(request):
+    """Fixture to convert the model once for the session."""
+    model_id = request.param
+    model_name = MODELS[model_id].get("name")
+    model_path = os.path.join(MODELS_DIR, model_name)
+    model_args = MODELS[model_id].get("convert_args", [])
+    logger.info(f"Preparing model: {model_name}")
+    # Convert the model if not already converted
+    if not os.path.exists(model_path):
+        logger.info(f"Converting model: {model_name}")
+        command = [
+            "optimum-cli", "export", "openvino",
+            "--model", model_name, model_path
+        ]
+        if model_args:
+            command.extend(model_args)
+        result = subprocess.run(command, check=True)
```

Comment:
HTTP errors are pretty common. Can you run this multiple times if fails

Reply:
Maybe even using a HF mirror for retry

Reply:
done